libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpH265Reader.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Assertions.checkNotNull;
+import static androidx.media3.common.util.Assertions.checkStateNotNull;
+import static androidx.media3.common.util.Util.castNonNull;
+
+import androidx.media3.common.C;
+import androidx.media3.common.ParserException;
+import androidx.media3.common.util.Log;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.Util;
+import androidx.media3.exoplayer.rtsp.RtpPacket;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.NalUnitUtil;
+import androidx.media3.extractor.TrackOutput;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.RequiresNonNull;
+
+/**
+ * Parses an H265 byte stream carried on RTP packets, and extracts H265 Access Units. Refer to
+ * RFC7798 for more details.
+ */
+/* package */ final class RtpH265Reader implements RtpPayloadReader {
+
+  private static final String TAG = "RtpH265Reader";
+
+  private static final long MEDIA_CLOCK_FREQUENCY = 90_000;
+
+  /** Offset of payload data within a FU type A payload. */
+  private static final int FU_PAYLOAD_OFFSET = 3;
+
+  /** Single Time Aggregation Packet type A. */
+  private static final int RTP_PACKET_TYPE_STAP_A = 48; // RFC7798 Section 4.4.2
+  /** Fragmentation Unit type A. */
+  private static final int RTP_PACKET_TYPE_FU_A = 49;
+
+  /** IDR NAL unit type. */
```

Comment:
I don't see IDR_W_LP in H265 spec or RFC7798, maybe I missed something? Or could it be IDR_W_RADL?

It'd be useful if a link is attached.